wa/workloads/facebook/uiauto/app/src/main/java/com/arm/wa/uiauto/facebook/UiAutomation.java

```diff
@@ -0,0 +1,190 @@
+/*    Copyright 2013-2015 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+
+package com.arm.wa.uiauto.facebook;
+
+import android.support.test.runner.AndroidJUnit4;
+import android.support.test.uiautomator.UiObject;
+import android.support.test.uiautomator.UiObjectNotFoundException;
+import android.support.test.uiautomator.UiSelector;
+
+import com.arm.wa.uiauto.BaseUiAutomation;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+@RunWith(AndroidJUnit4.class)
+public class UiAutomation extends BaseUiAutomation {
+
+    public static String TAG = "facebook";
+
+    /*
+     * The 'runWorkload' method implements the following activities
+     * Login to facebook account.
+     * Send a message.
+     * Check latest notification.
+     * Search particular user account and visit his/her facebook account.
+     * Update the facebook status
+     */
+    @Test
+    public void runWorkload() throws Exception {
```

Comment:
Could you break this up into several methods each performing a single logical step of execution, and the run method should be just a simple sequence of steps. That will make it a lot easier to maintain this.